# Computational Methods For Epitranscriptomics

<b>Course Overview</b>

<p align="justify">Epitranscriptome refers to all chemical modifications affecting cellular RNAs. Although more than 170 different types have been identified up to now, their punctual detection is yet  challenging. Profiling epitranscriptome modifications require ad hoc sequencing technologies as well as dedicated computational workflows. While Illumina RNAseq is the de facto  sequencing methodology for unveiling RNA editing events (not transient modifications), Nanopore direct RNA Sequencing has recently become one of the most powerful techniques to profile the whole repertoire of RNA modifications. Its capacity to directly sequence full length, native RNA molecules without the need for retrotranscription or amplification offers  remarkable advantages compared to other technologies based on short-read sequencing.  However, at the same time it poses various experimental and analytical challenges. 
This course provides a foundation on the experimental planning, analytical strategies and computational approaches for applying Nanopore direct RNA Sequencing to profile RNA  modifications. Through a combination of theoretical lectures and practical exercises, participants can learn the fundamental concepts and bioinformatics skills that will allow them  to start using Nanopore sequencing for epitranscriptome analysis. Additionally, the course also covers theoretical aspects of RNA editing as well as practical sessions focused on its  detection in human/mouse transcriptomes by Illumina RNAseq data. Participants can acquire specific skills in command-line tools (<a href="https://github.com/BioinfoUNIBA/REDItools">REDItools</a>) to call, annotate (<a href="http://srv00.recas.ba.infn.it/atlas/index.html">REDIportal</a>) and filter RNA  variants.</p>

<b>Course Material</b>

<b>Day 1</b>

- Invited Talk by Ivan Milenkovic (CRG Center - Novoa Group - Barcelona)

<i>"Decoding the epitranscriptome at single molecule resolution"</i> - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day1/Ivan_EvaMariaNovoa_ELIXIR.pdf">slides</a>

- Overview of transcriptomics and epitranscriptomics - Francesco Nicassio (IIT-Milan)

General overview on state-of-the-art approaches in transcriptome research - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day1/ELIXIR_course_Nicassio_Trancriptomics.pdf">slides</a>

- Profiling RNA modifications: concepts and methodologies - Mattia Pelizzola (IIT-Milan)

Theoretical introduction to epitranscriptomics and overview of the techniques to profile RNA modifications - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day1/RNAmod_methods_ELIXIR-1.pdf">slides</a>

- Introduction to Nanopore sequencing - Tommaso Leonardi (IIT-Milan)

General introduction to Nanopore sequencing providing a theoretical foundation on this technology - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day1/ELIXIR_course_Leonardi_Nanopore_Theory.pdf">slides</a>

<b>Day 2</b>

- Using Nanopore sequencing to detect RNA modifications - Camilla Ugolini  (IIT-Milan)

Theoretical introduction to the various tools and techniques to identify RNA modifications from Nanopore data - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day2/ELIXIR_course_Ugolini_Nanopore_analysis.pdf">slides</a>

- From raw data to RNA modifications: how the analysis works - Logan Malroney (IIT-Milan)

Introduction to the bioinformatics concepts, pipelines and software tools to analyse Nanopore data in order to detect RNA modifications - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day2/ELIXIR_course_Mulroney_%20Nanopore%20sequencing%20to%20detect%20RNA%20modifications.pdf">slides</a>

- Practical session - Logan Malroney (IIT-Milan)

Participants will analyse first-hand a Nanopore direct RNA Sequencing dataset, putting into practice what they learnt during the day - <a href="">data</a> - <a href="">scripts</a> 

- Introduction to RNA editing - Ernesto Picardi (UNIBA) 

General introduction to RNA editing - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day2/training-course-27-04-2023_picardi.pdf">slides</a>  

- RNA editing databases and REDIportal - Pietro D'Addabbo (UNIBA)

Introduction to biological databases for RNA editing and practice with REDIportal - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day2/REDIportal.ppt.pdf">slides</a>

- RNA editing detection by REDItools - Adriano Fonzino (UNIBA)

Theoretical and practical Introduction to the REDItools package - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day2/Epitransctiptomic_27_04_2023.pdf.PDF">slides</a> 

<b>Day 3</b>

- Using REDItools and REDIportal to profile RNA editing - Claudio Lo Giudice (UNIBA)

Theoretical introduction to computational workflows for identifying RNA editing events from  RNAseq data - <a href="">slides</a>

- De novo detection of RNA editing sites by RNAseq data - Claudio Lo Giudice (UNIBA)

Practical session in which participants will put hands on data and tools to detect de novo RNA editing sites in transcriptomic data - <a href="">data</a> <a href="">scripts</a>
 
- Profiling known RNA editing events in RNAseq experiments - Alessandro Silvestris (UNIBA)

Practical session in which participants will apply REDItools and REDIportal to analyse known  RNA editing events (including methods for differential RNA editing) - <a href="https://github.com/ELIXIR-IIB-training/computational_methods_for_epitrascriptomics/blob/main/day3/elixir_epitranscriptomics.pdf">slides</a>

