# Computational Methods For Epitranscriptomics

<b>Course Overview</b>

Epitranscriptome refers to all chemical modifications affecting cellular RNAs. Although more than 170 different types have been identified up to now, their punctual detection is yet  challenging. Profiling epitranscriptome modifications require ad hoc sequencing technologies as well as dedicated computational workflows. While Illumina RNAseq is the de facto  sequencing methodology for unveiling RNA editing events (not transient modifications), Nanopore direct RNA Sequencing has recently become one of the most powerful techniques to profile the whole repertoire of RNA modifications. Its capacity to directly sequence full length, native RNA molecules without the need for retrotranscription or amplification offers  remarkable advantages compared to other technologies based on short-read sequencing.  However, at the same time it poses various experimental and analytical challenges. 
This course provides a foundation on the experimental planning, analytical strategies and computational approaches for applying Nanopore direct RNA Sequencing to profile RNA  modifications. Through a combination of theoretical lectures and practical exercises, participants can learn the fundamental concepts and bioinformatics skills that will allow them  to start using Nanopore sequencing for epitranscriptome analysis. Additionally, the course also covers theoretical aspects of RNA editing as well as practical sessions focused on its  detection in human/mouse transcriptomes by Illumina RNAseq data. Participants can acquire specific skills in command-line tools (REDItools) to call, annotate (REDIportal) and filter RNA  variants.

Day 1: Wednesday 26th April 
14.00-14.30 Course introduction 
Introduction of the trainers/trainees and brief description of the program 
14.30-15.00 Talk by Eva Novoa 
Introductory epitranscriptomic talk by Eva Novoa (CRG Center - Barcelona) 
15.00-15.30 Overview of transcriptomics and epitranscriptomics 
General overview on state-of-the-art approaches in transcriptome research 
15.30-16.30 Profiling RNA modifications: concepts and methodologies Theoretical introduction to epitranscriptomics and overview of the techniques to profile RNA  modifications (antibody-based, MS, etc.) 
16.30-17.00 Coffee Break 
17.00-18.00 Introduction to Nanopore sequencing 
General introduction to Nanopore sequencing providing a theoretical foundation on this  technology 
Day 2: Thursday 27th April 
09.00-10.00 Using Nanopore sequencing to detect RNA modifications Theoretical introduction to the various tools and techniques to identify RNA modifications  from Nanopore data. 
10.00-11.00 From raw data to RNA modifications: how the analysis works Introduction to the bioinformatics concepts, pipelines and software tools to analyse  Nanopore data in order to detect RNA modifications 
11.00-11.30 Coffee Break 
11.30-13.00 Practical session 
Participants will analyse first-hand a Nanopore direct RNA Sequencing dataset, putting into  practice what they learnt during the day. 
13.00-15.00 Lunch Break 
15.00-15.30 Introduction to RNA editing 
General introduction to RNA editing  
15.30-16.30 RNA editing databases and REDIportal 
Introduction to biological databases for RNA editing and practice with REDIportal
16.30-17.00 Coffee Break 
17.00-18.00 RNA editing detection by REDItools 
Theoretical and practical Introduction to the REDItools package 
Day 3: Friday 28th April 
09.00-9.30 Using REDItools and REDIportal to profile RNA editing 
Theoretical introduction to computational workflows for identifying RNA editing events from  RNAseq data. 
9.30-11.00 De novo detection of RNA editing sites by RNAseq data Practical session in which participants will put hands on data and tools to detect de novo RNA editing sites in transcriptomic data. 
11.00-11.30 Coffee break 
11.30-13.00 Profiling known RNA editing events in RNAseq experiments Practical session in which participants will apply REDItools and REDIportal to analyse known  RNA editing events (including methods for differential RNA editing)

